### PR TITLE
Add @terrytangyuan to OWNERS

### PR DESCRIPTION
**Pull request by terrytangyuan**
I'll be mostly maintaining the docs for MPI as well as potentially adding docs for other operators we maintain.

### CODE REVIEW

File: OWNERS [hunk 1]
```diff
@@ -11,6 +11,7 @@ approvers:
   - pdmack
   - richardsliu
   - sarahmaddox
+  - terrytangyuan
```
Comment (sarahmaddox) [inline OWNERS]: Actually, we're planning to trim the number of approvers for the website, and focus on adding people as reviewers if they're looking after a specific Kubeflow component or section of the docs. The approvers will be limited to three or four people who have responsibility for the website as a whole. (I have a task set to fix this, and your PR is prompting me to get it done. :) )

Please would you remove your ID from the approvers section?

Comment (terrytangyuan) [reply OWNERS]: Got it. Done!

Comment (sarahmaddox) [reply OWNERS]: Thanks!

**Pull request merged**